src/web/app/components/question-types/question-constraint/constsum-recipients-question-constraint.component.ts

```diff
@@ -101,4 +101,43 @@ export class ConstsumRecipientsQuestionConstraintComponent implements OnInit {
 
     return set.size !== 1;
   }
+
+  get isAllPointsDistributed(): boolean {
+    return this.totalAnsweredPoints < this.totalRequiredPoints;
+  }
+
+  get isPointsOverAllocated(): boolean {
+    return this.totalAnsweredPoints > this.totalRequiredPoints;
+  }
+
+  get isWronglyAllUneven(): boolean {
+    return this.questionDetails.distributePointsFor === FeedbackConstantSumDistributePointsType.DISTRIBUTE_ALL_UNEVENLY
+        && !this.isAllPointsUneven;
+  }
+
+  get isCorrectlyAllUneven(): boolean {
+    return this.questionDetails.distributePointsFor === FeedbackConstantSumDistributePointsType.DISTRIBUTE_ALL_UNEVENLY
+        && this.isAllPointsUneven;
+  }
+
+  get isWronglySomeUneven(): boolean {
+    return this.questionDetails.distributePointsFor === FeedbackConstantSumDistributePointsType.DISTRIBUTE_SOME_UNEVENLY
+        && !this.isSomePointsUneven;
+  }
+
+  get isCorrectlySomeUneven(): boolean {
+    return this.questionDetails.distributePointsFor === FeedbackConstantSumDistributePointsType.DISTRIBUTE_SOME_UNEVENLY
+        && this.isSomePointsUneven;
+  }
+
+  get isSubmissionValid(): boolean {
```

Comment:
This is better suited as a interface which every *-constraint should override.

Reply:
This is your starting point should be where you fire the `isValid` event. Then your question becomes when should I fire the event? -> When the input changes? -> How can I know if the input changes?

Reply:
This seems quite complex, I think I will remove the front-end validation from this PR and work on it in another PR so the UAT fix can get pushed out first. Thanks for all the guidance!